client.go

```diff
@@ -165,29 +165,30 @@ func BuildUrl(requestURL string, genericParams GenericParams) string {
 }
 
 // NewRequest generates an http.Request for the API client to submit to Recurly.
-func (c *Client) NewRequest(method string, requestURL string, params *Params) (*http.Request, error) {
+func (c *Client) NewRequest(method string, requestURL string, params *Params, requestOptions *RequestOptions) (*http.Request, error) {
```

Comment:
I think it would be beneficial to write some comments on this function that address the oddities around params and requestOptions, and lay out the plan in the future to replace params